File: exchanges/bybit.py
import requests


def get_bybit_spot_trade_link(currency1: str, currency2: str) -> str:
    return f"https://www.bybit.com/ru-RU/trade/spot/{currency1}/{currency2}"


def get_bybit_futures_trade_link(currency1: str, currency2: str) -> str:
    return f"https://www.bybit.com/trade/usdt/{currency1}{currency2}"


def get_bybit_funding_rates() -> dict[str, dict[str, float]]:
    funding_rates = requests.get(
        "https://api.bybit.com/derivatives/v3/public/tickers?category=linear"
    )

    return {
        funding_rate["symbol"]: {
            "funding_rate": float(funding_rate["fundingRate"]) * 100,
            "index_price": float(funding_rate["indexPrice"]),
            "mark_price": float(funding_rate["markPrice"]),
            "next_funding_time": float(funding_rate["nextFundingTime"]),
            "predicted_funding_rate": "-",
        }
        for funding_rate in funding_rates.json()["result"]["list"]
        if "-" not in funding_rate["symbol"]
    }

File: exchanges/bybit.js
const axios = require('axios');

const { ExchangeName } = require('../constants');
const { formatFundingRate, getTimeString } = require('../utils');

class Bybit {
  getSpotTradeLink(symbol) {
    const currency = symbol.split('USDT')[0];
    return `https://www.bybit.com/ru-RU/trade/spot/${currency}/USDT`;
  }

  getFuturesTradeLink(symbol) {
    return `https://www.bybit.com/trade/usdt/${symbol}`;
  }

  async getFundingRates() {
    try {
      const { data: fundingRates } = await axios.get(
        'https://api.bybit.com/v5/market/tickers?category=linear'
      );

      return fundingRates.result.list
        .filter((fundingRate) => !fundingRate.symbol.includes('-'))
        .reduce((acc, fundingRate) => {
          const symbol = fundingRate.symbol;
          const nextFundingTime = parseInt(fundingRate.nextFundingTime);

          return {
            ...acc,
            [symbol.replace(/^10+/g, '')]: {
              fundingRate: formatFundingRate(fundingRate.fundingRate),
              indexPrice: fundingRate.indexPrice,
              markPrice: fundingRate.markPrice,
              nextFundingTime: getTimeString(nextFundingTime),
              fundingInterval: parseInt(fundingRate.fundingIntervalHour),
              predictedFundingRate: '-',
              spotLink: this.getSpotTradeLink(symbol.replace(/^10+/g, '')),
              futuresLink: this.getFuturesTradeLink(symbol),
              multiplier: symbol.match(/^10+/g)?.[0] ?? 1,
            },
          };
        });
    } catch (err) {
      console.log(`Ошибка получения данных фандинга ${ExchangeName.BYBIT}. ${err?.message}`);
    }
  }
}

module.exports = new Bybit();
